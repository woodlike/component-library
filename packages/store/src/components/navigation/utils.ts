export const getHeaderHeight = (scales: string[]): number[] =>
  scales.map(scale => parseInt(scale.replace(/[a-z]/gi, '')));
