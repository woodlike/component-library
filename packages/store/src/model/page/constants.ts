export const legalIds = [
    'imprint',
    'impressum',
    'protection',
    'datenschutzerklärung',
    'returns',
    'widerrufsbelehrung',
    'terms',
    'geschäftsbedingungen',
  ];