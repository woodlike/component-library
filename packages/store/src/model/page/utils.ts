import { filter, find, includes, map, toLower } from "ramda"

import { ShopifyNode } from "../../gatsby"

export const _hasNodeTitle = <T extends ShopifyNode>(ids: string[]) => (
  node: T,
) => {
  return !!find(id => includes(id, toLower(node.title)), ids)
}

export const _filterPageByTitle = <T extends ShopifyNode>(ids: string[]) => {
  // The filter operator is a curried function
  // It accepts an implicity second argument which accepts
  // an extended ShopifyNode array
  return filter<T>(_hasNodeTitle(ids))
}

export const _addPathToSlug = <T extends ShopifyNode>(pathname: string) => (
  node: T,
) => ({ ...node, slug: `/${pathname}/${node.handle}`.replace(/\/\/+/g, "/") })

export const _addSlugs = <T extends ShopifyNode>(
  callback: (node: T) => ShopifyNode,
  nodes: T[],
) => map(callback, nodes)
