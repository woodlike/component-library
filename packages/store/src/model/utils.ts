import { filter, find, includes, toLower } from 'ramda';

import { ShopifyNode } from '../gatsby';
import { ShopifyPage } from '.';

export const hasNodeTitle = <T extends ShopifyNode>(node: T, ids: string[]) =>
  !!find(id => includes(id, toLower(node.title)), ids);

export const filterPageByTitle = (ids: string[]) => filter<ShopifyPage>(node => hasNodeTitle(node, ids));

