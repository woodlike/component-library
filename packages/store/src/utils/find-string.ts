export function findString(strings: string[], search: string) {
  return !!strings.find(string => {
    const formattedString = string.toLowerCase().trim();
    const hyphen = /-/g;

    return !!search.search(hyphen) && formattedString.search(hyphen) === -1
      ? formattedString.replace(/\s/g, '') === search.replace(hyphen, '')
      : formattedString === search;
  });
}
