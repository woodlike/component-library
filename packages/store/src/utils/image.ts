import { ProductImageWidth, SourceSetProps } from '../gatsby';

function lazyAddWidths(sets: typeof ProductImageWidth) {
  const sizes = Object.keys(sets);
  const addWidth = (sources: SourceSetProps[]) =>
    sources.reduce((acc, set, idx) => acc.concat(`${set} ${sizes[idx]}w,`), '');
  return addWidth;
}

export const addSrcsetDescriptor = lazyAddWidths(ProductImageWidth);

export function createSrcSets(srcSets: SourceSetProps[]): string {
  const sizes = Object.keys(ProductImageWidth);
  return srcSets.reduce(
    (acc, set, idx) => acc.concat(`${set} ${sizes[idx]}w,`),
    '',
  );
}