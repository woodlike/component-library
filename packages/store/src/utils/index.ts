export * from './generate-id';
export * from './price-formatter';
