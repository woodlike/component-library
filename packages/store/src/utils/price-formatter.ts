export function priceFormatter(
  srcPrice: string,
  locale: string,
  currency: string,
): string {
  return parseFloat(srcPrice).toLocaleString(locale, {
    style: 'currency',
    currency: currency,
  });
}
