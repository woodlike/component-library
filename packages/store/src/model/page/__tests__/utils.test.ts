import { ShopifyPage } from ".."
import { _filterPageByTitle } from "../utils"
import { legalIds } from "../constants"
import { shopifyPages } from "../../../../__mocks__"

describe("Pages Utils", () => {
  it("should return a filterd list of legal page nodes - (filterPageByTitle)", () => {
    const filterLegalPages = _filterPageByTitle(legalIds)
    const pages = shopifyPages.nodes as ShopifyPage[]

    expect(filterLegalPages(pages)).toMatchSnapshot()
  })
})
