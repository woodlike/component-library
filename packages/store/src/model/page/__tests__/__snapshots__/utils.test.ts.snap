// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Utils should return a filterd list of legal page nodes - (filterPageByTitle) 1`] = `
Array [
  Object {
    "handle": "terms-conditions",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzEyOTAxNjkwMA==",
    "title": "Standard Business Terms and customer information",
  },
  Object {
    "handle": "privacy-policy",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzEyOTIxNTc0OA==",
    "title": "Data protection declaration",
  },
  Object {
    "handle": "agb",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzEyOTIxNTk0MA==",
    "title": "Allgemeine Geschäftsbedingungen und Kundeninformationen",
  },
  Object {
    "handle": "impressum",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzEyOTIxNjA2OA==",
    "title": "Impressum",
  },
  Object {
    "handle": "datenschutzerklarung",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzEyOTIxNjE5Ng==",
    "title": "Datenschutzerklärung",
  },
  Object {
    "handle": "cancellation-right",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzEzMDAwODc3Mg==",
    "title": "Returns & Exchanges",
  },
  Object {
    "handle": "imprint",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzEzMDA2MTEyNA==",
    "title": "About Us - Imprint",
  },
  Object {
    "handle": "widerrufsbelehrung",
    "id": "Shopify__Page__Z2lkOi8vc2hvcGlmeS9QYWdlLzI1MDc1MTMxMw==",
    "title": "Widerrufsbelehrung",
  },
]
`;
